app/javascript/packs/articlePage.jsx

```diff
@@ -71,6 +72,31 @@ function showAnnouncer() {
   document.getElementById('article-copy-link-announcer').hidden = false;
 }
 
+// Temporary Ahoy Stats for comment section clicks
+function trackCommentSectionClicks() {
+  document.getElementById('comments')?.addEventListener('click', (event) => {
+    ahoy.track('Comment section click', {
+      page: location.href,
+      element: event.target.id,
+    });
+  });
+}
+
+// Temporary Ahoy Stats for scrolling to comments section
+function trackCommentSectionScroll() {
+  const callback = (entries, _observer) => {
+    entries.forEach((entry) => {
+      if (entry.isIntersecting) {
```

Comment:
This would be the case for small (in height) posts, right? Maybe guarding this with (or something long the lines of) would help:

```js
if (document.querySelector('.crayons-article__main').scrollHeight > window.innerHeight) {
 ....
}
```

This ^ means that only posts that are "taller than the viewport" would be tracked (when the page has scrolled to the comment section). Short posts wouldn't be considered because they already appear on screen on page load. False positives would be a little worse than ignoring these because it would inflate the numbers IMHO

Reply:
It looks like there's an interplay here:

Did the person "see the comments section" and did they click in it (based on the above).

Reply:
Would it make sense to name it "Seen the comment section"?

Reply:
Also, would it be good to keep track (even if it is on a temporary window variable) of the posts that have been tracked, or remove the observer once it has tracked the event?

I haven't tested this locally, but I'm thinking there's a chance one user scrolling could add more than one event per post, which may not be ideal 🤔

Reply:
> Would it make sense to name it "Seen the comment section"?

@amywtlin What do you care about here? Do you want to track _scrolling_ to the comments section, or just seeing it?

Reply:
I care more about if comments section has been visible to the user, no matter how they get there. So seeing it would be good enough.

Reply:
Thanks @amywtlin: will rename this to reflect it's tracking that the comments section was displayed/seen

Reply:
Oh very cool @aitchiss, thanks!